eng/tools/rush-runner.js

```diff
@@ -28,7 +34,78 @@ const parseArgs = () => {
       }
     }
   }
-  return [baseDir, action, services, flags];
+  return [baseDir, action, services, flags, buildTransitiveDep];
+};
+
+const getAllPackageJsonPaths = (baseDir) => {
+  // Find and return path to all packages in repo
+  const packagePaths = [];
+  const serviceDirs = fs.readdirSync(path.resolve(path.join(baseDir, 'sdk')))
+    .filter(f => !f.startsWith('.'))
+    .map(f => path.resolve(path.join(baseDir, 'sdk', f)));
+
+  for (const serviceDir of serviceDirs) {
+    for (const pkgPath of getPackageJsons(serviceDir))
+      packagePaths.push(pkgPath);
+  }
+  return packagePaths;
+};
+
+const getPackageGraph = (baseDir) => {
+  // Create a graph of packages with edges to packages that are dependent
+  // for e.g.  C requires A and B and D requires A and C
+  // Graph is as follows
+  // { A: [C, D], B: [C], C: [D]}
+  let packageGraph = new Map();
+  const packageJsons = getAllPackageJsonPaths(baseDir);
+
+  for (const filePath of packageJsons) {
+    const contents = JSON.parse(fs.readFileSync(filePath, 'utf8'));
+    const pkgName = contents["name"];
+    if (!contents.hasOwnProperty('dependencies'))
+      continue;
+
+    const dependencies = Object.keys(contents['dependencies']).filter(f => f.startsWith("@azure"));
```

Comment:
Should we do "@azure"? Could we instead just look to see if the project exists in the rush workspace?